src/Form/CloudFlareAdminSettingsForm.php

```diff
@@ -65,12 +144,28 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
     $email = $form_state->getValue('email');
-    $is_email_valid = \Drupal::service('email.validator')->isValid($email);
+    $apikey = $form_state->getValue('apikey');
+    $is_email_valid = $this->emailValidator->isValid($email);
+
+    // Using the bare-metal-class here for diagnostic purposes.
+    $zone_api_direct = new ZoneApi($apikey, $email);
 
     if (!$is_email_valid) {
       $form_state->setErrorByName('email', $this->t('Please enter a valid e-mail address.'));
     }
 
+    try {
+      // Simply using this call to confirm that the creds can authenticate
+      // against the CloudFlareApi. An exception here tell us the creds are
+      // invalid.
+      $zone_api_direct->listZones();
+      $this->state->incrementApiRateCount();
```

Comment:
Just thinking architecture here, wouldn't it be a better idea to move the rate counting maintenance deeper into your SDK? Now I do understand this could be quite a bit of work, not worth it if CF is about to fix this weird limitation.